core/src/databases.rs

```diff
@@ -6,9 +6,11 @@ use bson::Bson;
 #[derive(Debug)]
 pub struct MongoDatabases {
     // The list of all the databases
-    databases_names: Vec<String>,
+    database_names: Vec<String>,
     // The current database index.
     current_db_index: u32,
```

Comment:
Is there any reason we can't make `current_db_index` a `usize` and not store the length here? Seems we have some unnecessary casting (and storing an extra 4 bytes), but maybe I'm missing something?

Reply:
That's a good point.  It's the type that was in the design, I took a look and it doesn't seem to need to remain `u32`, switching it to `usize` and removing the `current_db_index` variable from the struct.